src/test/java/org/apache/bcel/classfile/JDKClassDumpTestCase.java

```diff
@@ -81,7 +80,8 @@ private void compare(final JavaClass jc, final InputStream inputStream, final St
             int i = 0;
             for (final int out : baos.toByteArray()) {
                 final int in = src.read();
-                assertEquals(in, out & 0xFF, name + ": Mismatch at " + i);
+                int j = i;
```

Comment:
Do we really need `j`?

Reply:
Yes - values in a lambda expression must be final or effectively final, and `i` isn't, so I assigned it to `j`, which is (since it's scopped in the loop's body).

Alternatively, we can leave the message as is and not replace it with a `Supplier`.